File: code/go/ch08/src/luago/state/api_set.go
package state

// [-2, +0, e]
// http://www.lua.org/manual/5.3/manual.html#lua_settable
func (self *luaState) SetTable(idx int) {
	t := self.stack.get(idx)
	v := self.stack.pop()
	k := self.stack.pop()
	self.setTable(t, k, v)
}

// [-1, +0, e]
// http://www.lua.org/manual/5.3/manual.html#lua_setfield
func (self *luaState) SetField(idx int, k string) {
	t := self.stack.get(idx)
	v := self.stack.pop()
	self.setTable(t, k, v)
}

// [-1, +0, e]
// http://www.lua.org/manual/5.3/manual.html#lua_seti
func (self *luaState) SetI(idx int, i int64) {
	t := self.stack.get(idx)
	v := self.stack.pop()
	self.setTable(t, i, v)
}

// t[k]=v
func (self *luaState) setTable(t, k, v luaValue) {
	if tbl, ok := t.(*luaTable); ok {
		tbl.put(k, v)
		return
	}

	panic("not a table!")
}

File: code/csharp/ch07/luavm/state/ApiSet.cs
using System;

namespace luavm.state
{
    partial struct LuaState
    {
        public void SetTable(int idx)
        {
            var v = stack.pop();
            var k = stack.pop();
            stack.setTable(idx, new LuaValue(k), new LuaValue(v));
        }

//        internal LuaTable setTable(object t, LuaValue k, LuaValue v)
//        {
//            if (new LuaValue(t).isLuaTable())
//            {
//                var tbl = (LuaTable) t;
//                tbl.put(k, v);
//                return tbl;
//            }
//
//            throw new Exception("not a table!");
//        }

        public void SetField(int idx, string k)
        {
            var t = stack.get(idx);
            var v = stack.pop();
            stack.setTable(idx, new LuaValue(k), new LuaValue(v));
        }

        public void SetI(int idx, long n)
        {
            var t = stack.get(idx);
            var v = stack.pop();
            stack.setTable(idx, new LuaValue(n), new LuaValue(v));
        }
    }
}